sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -964,24 +972,7 @@ private static BlobRequestConditions setEtag(BlobRequestConditions requestCondit
         AtomicLong totalProgress) {
 
         // Extract the body.
-        Flux<ByteBuffer> data = response.getValue()
-            .flatMap(buffer -> {
-                /*
-                 * Buffer the data contained in the 'ByteBuffer' as it passes through the stream. This resolves an issue
-                 * where Reactor Netty begins to release the underlying 'ByteBuf' after the on next operations have
-                 * completed, the release buffer may be overwritten before writing to file has completed which leads to
-                 * corrupted files.
-                 */
-                int offset = buffer.position();
-                int size = buffer.remaining();
-                byte[] duplicate = new byte[size];
-
-                for (int i = 0; i < size; i++) {
-                    duplicate[i] = buffer.get(i + offset);
-                }
-
-                return Mono.just(ByteBuffer.wrap(duplicate));
-            });
+        Flux<ByteBuffer> data = StorageImplUtils.deepCloneStreamBuffers(response.getValue());
```

Comment:
I think we don't need this copy anymore for files. Since there's already a deep copy in files, if we do a deep copy in this method, we'll be doing one too many copies, right?

Reply:
Removed